app/models/manageiq/providers/vmware/discovery.rb

```diff
@@ -4,70 +4,59 @@ module ManageIQ
   module Providers
     module Vmware
       class Discovery
-        SERVER_CONSOLE_PORTS = [902, 912].freeze
-
-        ESX_PORTS = [902, 903].freeze
-
-        VC_PORTS = [
-          [
-            135, # VC < 5.1 or
-            7444 # VC >= 5.1
-          ],
-          [
-            139,  # VC < 5.1 or
-            2012, # VC >= 5.1
-            2013,
-            2014
-          ]
-        ].freeze
-
         def self.probe(ost)
-          # Check if VMware Server
-          ost.hypervisor << :vmwareserver if ManageIQ::NetworkDiscovery::Port.scan_open(ost, SERVER_CONSOLE_PORTS).length == 2
+          # Get Info from VMware (vSphere) Web Service API
+          info = retrieve_webservice_info(ost.ipaddr)
 
```

Comment:
How about `return if info.nil?` here and not need to do `info&.productLineId` and `info&.osType` below?

Reply:
I saw codeclimate only few times, but doesn't it evaluate as higher cognitive complexity when using return in middle of method?

Reply:
If the code makes more sense I wouldn't worry about what codeclimate says.  If you'd rather one return you can raise in `retrieve_webservice_info` if `vim.about` is nil